src/api_manager/check_auth.cc

```diff
@@ -170,7 +171,14 @@ void AuthChecker::Check() {
 void AuthChecker::GetAuthToken() {
   Request *r = context_->request();
   std::string auth_header;
+
   if (!r->FindHeader(kAuthHeader, &auth_header)) {
+    // IAP header is of format "X-Goog-Iap-Jwt-Assertion": "eyJhbG...". No
+    // "Bearer" prefix is needed.
+    if (r->FindHeader(kAuthHeaderIAP, &auth_header)) {
+      auth_token_ = auth_header;
+      return;
+    }
```

Comment:
Let's put the check for IAP header before checking "authorization" header. It's possible that IAP header and authorization header both exist (customer use authorization header for their own auth purpose). In that case, ESP will just check auth token in IAP header.